### PR TITLE
test(snapshot): add e2e test case for online volume snapshot

### LINKED ISSUE #101333: Add online snapshot e2e
#### What would you like to be added:
Ref: https://github.com/kubernetes/kubernetes/pull/101280/files#r616971565

Currently our snapshot tests only test offline snapshots, taken when after the volume is unmounted. We are missing coverage for online snapshots, taken when the volume is mounted.

#### Why is this needed:
I think online snapshots is the more common scenario

### PR DESCRIPTION
**Pull request by hoyho**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add online snapshot e2e test as there is missing coverage for non-ephemeral volumes

#### Which issue(s) this PR fixes:
Fixes #101333

#### Special notes for your reviewer:
This PR covers online snapshot scenario
Cloning related tests are not included. Maybe will submit later .

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/testsuites/snapshottable.go [hunk 1]
```diff
@@ -396,6 +396,93 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
...
+					pvcNamespace := pod.Namespace
+					pvc, err = cs.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(ctx, pvcName, metav1.GetOptions{})
+					framework.ExpectNoError(err, "get PVC")
+				} else {
+					pvc = volumeResource.Pvc
+				}
+
+				ginkgo.By("creating a snapshot from the source PVC when the pod is using it")
+				parameters := map[string]string{}
+				sr := storageframework.CreateSnapshotResource(ctx, sDriver, config, pattern, pvc.Name, pvc.Namespace, f.Timeouts, parameters)
```
Comment (xing-yang) [inline test/e2e/storage/testsuites/snapshottable.go:424]: See this comment: https://github.com/kubernetes/kubernetes/pull/101280/files#r616971565

Comment (hoyho) [reply test/e2e/storage/testsuites/snapshottable.go:424]: > See this comment: https://github.com/kubernetes/kubernetes/pull/101280/files#r616971565

> I think the original test is considered more like offline test. The pod was deleted first and then snapshot is taken. The subtle thing here is that even when pod is deleted from api server, there might be a chance that volume is not finished unstaging yet.

So the new online test need to change this logic.

Hi @xing-yang  Do you mean this comment? 
It makes me confused. The snapshot is taken when the pod is running. What do we need to check here?

**Pull request closed without merging**